bot/exts/utilities/wtf_python.py

```diff
@@ -0,0 +1,122 @@
+
+
+class WTFPython(commands.Cog):
+    """Cog that allows getting WTF Python entries from the WTF Python repository."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.headers: dict[str, str] = {}
+        self.last_fetched = datetime.datetime.now()
+        self.fetch_readme.start()
+
+    @tasks.loop(minutes=README_REFRESH)
+    async def fetch_readme(self) -> None:
+        """Gets the content of README.md from the WTF Python Repository."""
+        refresh = self.last_fetched + datetime.timedelta(minutes=README_REFRESH)
+
+        if refresh > datetime.datetime.now() and self.headers:
+            return  # Cache should be up-to-date
+
+        async with self.bot.http_session.get(f"{WTF_PYTHON_RAW_URL}README.md") as resp:
+            log.trace("Fetching the latest WTF Python README.md")
+            if resp.status == 200:
+                raw = await resp.text()
+                self.parse_readme(raw)
+
+    def parse_readme(self, data: str) -> None:
+        """
+        Parses the README.md into a dict.
+
+        It parses the readme into the `self.headers` dict,
+        where the key is the heading and the value is the
+        link to the heading.
+        """
+        # Match the start of examples, until the end of the table of contents (toc)
+        table_of_contents = re.search(
+            r"\[👀 Examples\]\(#-examples\)\n([\w\W]*)<!-- tocstop -->", data
+        )[0].split("\n")
+
+        for header in list(map(str.strip, table_of_contents)):
+            match = re.search(r"\[▶ (.*)\]\((.*)\)", header)
+            if match:
+                self.headers[match[0]] = f"{BASE_URL}{match[1]}"
+
+    def fuzzy_match_header(self, query: str) -> Optional[str]:
+        """
+        Returns the fuzzy match of a query if its ratio is above "MINIMUM_CERTAINTY" else returns None.
+
+        "MINIMUM_CERTAINTY" is the lowest score at which the fuzzy match will return a result.
+        The certainty returned by rapidfuzz.process.extractOne is a score between 0 and 100,
+        with 100 being a perfect match.
+        """
+        match, certainty, _ = rapidfuzz.process.extractOne(query, self.headers.keys())
+        return match if certainty > MINIMUM_CERTAINTY else None
+
+    @commands.command(aliases=("wtf", "WTF"))
+    async def wtf_python(self, ctx: commands.Context, *, query: str) -> None:
+        """
+        Search WTF Python repository.
+
+        Gets the link of the fuzzy matched query from https://github.com/satwikkansal/wtfpython.
+        Usage:
+            --> .wtf wild imports
+        """
+        match = self.fuzzy_match_header(query)
+        if match:
+            embed = Embed(
+                title=f"WTF Python Search Result For {query}",
+                colour=constants.Colours.dark_green,
+                description=f"[Go to Repository Section]({self.headers[match]})",
+            )
+            embed.set_thumbnail(url=f"{WTF_PYTHON_RAW_URL}images/logo.png")
```

Comment:
This looks poor on dark mode:

![image](https://user-images.githubusercontent.com/38372706/135146203-33ea193d-0f73-42cd-b30a-64109de34bdc.png)

Reply:
Since the image is a .png, how can I change the embed background but only for dark mode?

Reply:
You'll have to find a picture that works better on both platforms, there's no way to differentiate it like that.

Reply:
I could save it as a .jpg and add it to the resources folder?

Reply:
Because of the license ([see here](https://github.com/satwikkansal/wtfpython/blob/master/LICENSE)) we actually could do that afaik.

For legal reasons I am not impliying anything and this is not legal advice ;)

Reply:
How about no image at all?

Reply:
Doesn't look to bad to me:

![image](https://user-images.githubusercontent.com/42116429/135197578-a981403c-a1ee-45a1-a846-3dc654207cda.png)